File: Java/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indexes = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; ++i){
            if(indexes.get(target - nums[i]) != null){
                return new int[]{indexes.get(target - nums[i]), i};
            }
            indexes.put(nums[i], i);
        }
        return null;
    }
}

File: Python/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i in range(len(nums)):
            offset = target - nums[i]
            if offset in dict:
                return [i, dict[offset]]
            dict[nums[i]] = i
        return []
        